assets/shiny-server.js

```diff
@@ -93,14 +93,12 @@
           } catch (e) {
             return false;
           }
-        }
+        };
 
-        var availableOptions = ['websocket', 'xdr-streaming', 'xhr-streaming', 
-            'iframe-eventsource', 'iframe-htmlfile', 'xdr-polling', 
-            'xhr-polling', 'iframe-xhr-polling', 'jsonp-polling'];
+        var availableOptions = {{{protocols}}}; 
```

Comment:
Why not build the array when rendering the template like we do elsewhere? Something like:
```
var protocols = [ {{#each data}}'{{this}}'{{#unless @last }}, {{/unless}}{{/each}} ];
```
(this might not work with all versions of handlebars)